Client/Frontend/Browser/TabManager.swift

```diff
@@ -251,12 +251,21 @@ class TabManager : NSObject {
 
 extension TabManager {
     class SavedTab: NSObject, NSCoding {
-        let url: NSURL?
+        var urls = [NSURL]()
         let isSelected: Bool
         let screenshot: UIImage?
+        let currentPage: Int
 
         init?(browser: Browser, isSelected: Bool) {
-            self.url = browser.url
+            let currentItem = browser.webView?.backForwardList.currentItem
```

Comment:
We still need to handle the double-restore case where tabs are restored with their `urls`, but that tab never loads. To do that, you can see if the tab's `SessionData` is non-`nil` here; if so, just use it directly. If it is `nil`, look through the back/forward list like you do below.